client/components/sidebar/vrackSection/vrack-section-sidebar.service.js

```diff
@@ -11,9 +11,25 @@ class VrackSectionSidebarService {
                 type: "VRACK"
             }
         ];
+        //vRacks defer object
+        this.vRacks = $q.defer();
+    }
+
+    /**
+     * returns promise that will be resolved to list of vRacks of current user
+     *
+     * @returns promise that will be resolved to an array of vRack services available for current logged in user
+     * @memberof VrackSectionSidebarService
+     */
+    getVracks() {
+        return this.vRacks.promise;
     }
 
     fillSection (services) {
+        if(services && _.isArray(services)) {
```

Comment:
I don't think the sidebarMenu should be responsible to save users' vrack for later use.

Reply:
from where I can get vracks loaded as part of loading left side menu? I did not find where they are cached so I added to cache in this service.